dotnet-desktop-guide/net/wpf/properties/property-value-inheritance.md

```diff
@@ -0,0 +1,75 @@
+
+## Making a custom property inheritable
+
+You can make a custom dependency property inheritable by enabling the <xref:System.Windows.FrameworkPropertyMetadata.Inherits> property in an instance of <xref:System.Windows.FrameworkPropertyMetadata>, and then registering your custom dependency property with that metadata instance. By default, `Inherits` is set to `false` in `FrameworkPropertyMetadata`. Making a property value inheritable affects performance, so only set `Inherits` to `true` if that feature is needed.
+
+When you register a dependency property with `Inherits` enabled in metadata, use the <xref:System.Windows.DependencyProperty.RegisterAttached%2A> method as described in [Register an attached property](how-to-register-an-attached-property.md). Also, assign a default value to the property so that an inheritable value exists. You might also want to create a property wrapper with `get` and `set` accessors on the owner type, just as you would for a nonattached dependency property. That way you can set the property value using the property wrapper on an owner or derived type. The following example creates a dependency property named `IsTransparent`, with `Inherits` enabled and a default value of `false`. The example also includes a property wrapper with `get` and `set` accessors.
+
+:::code language="csharp" source="./snippets/property-value-inheritance/csharp/MainWindow.xaml.cs" id="RegisterAttachedPropertyWithValueInheritance":::
+:::code language="vb" source="./snippets/property-value-inheritance/vb/MainWindow.xaml.vb" id="RegisterAttachedPropertyWithValueInheritance":::
+
+Attached properties are conceptually similar to global properties. You can check their value on any <xref:System.Windows.DependencyObject> and get a valid result. The typical scenario for attached properties is to set property values on child elements, and that scenario is more effective if the property in question is implicitly present as an attached property on each <xref:System.Windows.DependencyObject> element in the tree.
+
+## Inheriting property values across tree boundaries
+
+Property inheritance works by traversing a tree of elements. This tree is often parallel to the logical tree. However, whenever you include a WPF core-level object, such as a <xref:System.Windows.Media.Brush>, in the markup that defines an element tree, you've created a discontinuous logical tree. A true logical tree doesn't conceptually extend through the `Brush`, because the logical tree is a WPF framework-level concept. You can use the helper methods of <xref:System.Windows.LogicalTreeHelper> to analyze and view the extent of a logical tree. Property value inheritance is able to pass inherited values through a discontinuous logical tree, but only if the inheritable property was registered as an attached property and there isn't a deliberate inheritance-blocking boundary, such as a <xref:System.Windows.Controls.Frame>.
```

Comment:
The statement here (and in the original article) implies that you can create inherited properties through a non-attached DP:

> Property value inheritance is able to pass inherited values through a discontinuous logical tree, but only if the inheritable property was registered as an attached property and ...

There is a note in [Inherits property](https://review.docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkpropertymetadata.inherits?branch=main&view=windowsdesktop-6.0#System_Windows_FrameworkPropertyMetadata_Inherits), in the remarks section, that discourages this design. I think it would be good to have here too:

> Although property value inheritance might appear to work for nonattached dependency properties, the inheritance behavior for a nonattached property through certain object-object divisions in the runtime tree is undefined. Always use RegisterAttached to register properties where you specify Inherits in the metadata.

Reply:
@adegio, thanks for looking into that VB issue. I've commented a vb code line that references the x:Name value, once that bug is resolved we can uncomment.

The **Inheriting property values across tree boundaries** section already has the note you asked for (see screenshot below). Also, the text does say (see bold): "Property value inheritance is able to pass inherited values through a discontinuous logical tree, **but only if the inheritable property was registered as an attached property** and ...". So, I think we've got this covered if I'm understanding you correctly.

![image](https://user-images.githubusercontent.com/86677757/148299109-f7903226-03c7-4bab-aaa2-f9932a6e6e4d.png)

Reply:
AHAHAH whoops on the note, I thought I was looking at something else and the note was missing here. Doh! Too many tabs!